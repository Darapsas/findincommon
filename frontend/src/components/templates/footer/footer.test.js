import React from "react";
import Footer from "./footer";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("<Footer />", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Footer text should contain application name "Find In Common"', () => {
    const wrapper = shallow(<Footer />);
    expect(
      wrapper
        .find(".footer-text")
        .render()
        .text()
    ).toMatch(/Find In Common/);
  });

  it("Footer should contain a button that scrolls to page starting position", () => {
    const wrapper = shallow(<Footer />);
  });
});
