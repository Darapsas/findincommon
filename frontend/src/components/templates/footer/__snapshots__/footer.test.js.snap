// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches the snapshot 1`] = `
<footer
  className="text-muted fixed-bottom"
>
  <div
    className="container flex-v-center"
  >
    <span
      className="footer-text"
    >
      Find In Common is made and maintained by Â© Darius Rainys 2019
    </span>
    <span
      className="float-right"
    >
      <button
        className="btn btn-sm btn-outline-secondary"
        onClick={[Function]}
        type="button"
      >
        Back to top
      </button>
    </span>
  </div>
</footer>
`;
